File: javaee/ServletStudy/src/main/java/com/ung8023/foundation/java/ServletImplementsServlet.java
package com.ung8023.foundation.java;

import javax.servlet.*;
import java.io.IOException;

/**
 * 通过实现Servlet接口创建Servlet
 */
public class ServletImplementsServlet implements Servlet {

    ServletConfig servletConfig;

    @Override
    public void init(ServletConfig config) throws ServletException {
        System.out.println("init");
        servletConfig = config;
    }

    @Override
    public ServletConfig getServletConfig() {
        return servletConfig;
    }

    @Override
    public void service(ServletRequest req, ServletResponse res) throws ServletException, IOException {
        System.out.println("service");
        res.getWriter().write("This is a Servlet");
    }

    @Override
    public String getServletInfo() {
        return "通过实现Servlet接口创建一个Servlet";
    }

    @Override
    public void destroy() {
        System.out.println("servlet销毁了");
    }
}

File: javaee/ServletStudy/src/main/java/com/ung8023/foundation/kotlin/ServletImplementsServlet.kt
package com.ung8023.foundation.kotlin

import javax.servlet.Servlet
import javax.servlet.ServletConfig
import javax.servlet.ServletRequest
import javax.servlet.ServletResponse

class ServletImplementsServlet: Servlet{

    private var servletConfig: ServletConfig? = null

    override fun getServletConfig(): ServletConfig? {
        return servletConfig
    }

    override fun destroy() {
        println("销毁了")
    }

    override fun init(config: ServletConfig?) {
        servletConfig = config
    }

    override fun getServletInfo(): String = "ServletImplementsServletKt"

    override fun service(req: ServletRequest?, res: ServletResponse?) {
        res?.writer?.write("这是一个Kotlin的Servlet")
    }
}